syntax = "proto3";

service UserService {
  rpc GetAllUsers (Empty) returns (UserList);
  rpc GetUser (UserRequest) returns (UserResponse);
  rpc CreateUser (CreateUserRequest) returns (UserResponse);
  rpc UpdateUser (UpdateUserRequest) returns (UserResponse);
  rpc DeleteUser (UserRequest) returns (DeleteResonse);
}

message User{
  string id=1;
  string name=2;
  string email=3;
  string created_at=4;
}
message Empty{}

message UserRequest {
    string id=1;
}
message CreateUserRequest{
  string name=1;
  string email=2;
}
message UpdateUserRequest{
  string id=1;
  string name=2;
  string email=3;
}

message UserResponse {
    bool success = 1;
    string message = 2;
    User user = 3;   
}

message UserList{
    bool success = 1;
    repeated User users = 2;
    int32 count = 3;
}

message DeleteResonse{
    bool success=1;
    string message=2;
}